src/test/java/com/github/checkstyle/patchfilter/SuppressionPatchFilterTest.java

```diff
@@ -51,33 +58,166 @@ public void testAccept() throws Exception {
+        final UnifiedDiffFile file2 = diff.getFiles().get(1);
+        assertEquals("diff -U0 old/f2 new/f2", file2.getDiffCommand());
+        // same issue here is f2 not old/f2 and new/f2
+        assertEquals("f2", file2.getFromFile());
+        assertEquals("f2", file2.getToFile());
+        assertEquals(1, file2.getPatch().getDeltas().size());
+        assertNull(diff.getTail());
+    }
+
+    @Test
+    public void testMultiChangedFilesOnOnePatchTwo() throws Exception {
         final String fileName = getPath("MultiChangedFilesOnOnePatch.txt");
-        final SuppressionPatchFilter filter = createSuppressionPatchFilter(fileName);
-        final LocalizedMessage message1 = new LocalizedMessage(7, 1, null, "msg", null,
-                SeverityLevel.ERROR, null, getClass(), null);
-        final AuditEvent ev1 = new AuditEvent(this, "Test2.java", message1);
-        assertTrue(filter.accept(ev1),
-                "Audit event should be rejected when there are no matching patch filters");
-        final LocalizedMessage message2 = new LocalizedMessage(77, 1, null, "msg", null,
-                SeverityLevel.ERROR, null, getClass(), null);
-        final AuditEvent ev2 = new AuditEvent(this, "Test1.java", message2);
-        assertFalse(filter.accept(ev2),
-                "Audit event should be rejected when there are no matching patch filters");
-        final LocalizedMessage message3 = new LocalizedMessage(7, 1, null, "msg", null,
-                SeverityLevel.ERROR, null, getClass(), null);
-        final AuditEvent ev3 = new AuditEvent(this, "Test1.java", message1);
-        assertTrue(filter.accept(ev1),
-                "Audit event should be rejected when there are no matching patch filters");
+        final FileInputStream ins = new FileInputStream(fileName);
+        final UnifiedDiff diff = UnifiedDiffReader.parseUnifiedDiff(ins);
+        assertEquals(2, diff.getFiles().size());
+        final UnifiedDiffFile file1 = diff.getFiles().get(0);
+        assertEquals("diff -u Origin/Test1.java Update/Test1.java",
+                file1.getDiffCommand());
+        assertEquals("Origin/Test1.java", file1.getFromFile());
+        assertEquals("Update/Test1.java", file1.getToFile());
+        assertEquals(1, file1.getPatch().getDeltas().size());
+        final UnifiedDiffFile file2 = diff.getFiles().get(1);
+        assertEquals("diff -u Origin/Test2.java Update/Test2.java",
+                file2.getDiffCommand());
+        assertEquals("Origin/Test2.java", file2.getFromFile());
+        assertEquals("Update/Test2.java", file2.getToFile());
+        assertEquals(1, file2.getPatch().getDeltas().size());
+        assertNull(diff.getTail());
+    }
+
+    @Test
+    public void testGitDiffPatch() throws Exception {
+        final String fileName = getPath("PatchFileFromDiffTools/GitDiffPatch.txt");
+        final FileInputStream ins = new FileInputStream(fileName);
+        final UnifiedDiff diff = UnifiedDiffReader.parseUnifiedDiff(ins);
```

Comment:
Please extract these 3 lines to a new method `parseUnifiedDiff(string)`. This is used in every unit test here.

Reply:
done